epp-client/src/connection/client.rs

```diff
@@ -97,35 +96,34 @@ impl EppClient {
             ext_uris.map(|uris| uris.iter().map(|u| u.to_string()).collect::<Vec<String>>());
 
         let connection = EppConnection::new(registry.to_string(), stream).await?;
-        EppClient::build(connection, credentials, ext_uris).await
-    }
-
-    /// Makes a login request to the registry and initializes an EppClient instance with it
-    async fn build(
-        connection: EppConnection,
-        credentials: (String, String),
-        ext_uris: Option<Vec<String>>,
-    ) -> Result<EppClient, Box<dyn Error>> {
-        let mut client = EppClient {
+        Ok(EppClient {
             connection,
             credentials,
             ext_uris,
-            // client_tr_id_fn: Arc::new(default_client_tr_id_fn),
-        };
+        })
+    }
 
-        let client_tr_id = generate_client_tr_id(&client.credentials.0)?;
+    /// Makes a login request to the registry
+    pub async fn login(&mut self, client_tr_id: &str) -> Result<EppLogoutResponse, error::Error> {
```

Comment:
I feel like at this point we shouldn't provide this method (or `logout()`) at all, and just make the caller construct  an `EppLogin`? Then the `EppClient` no longer needs `credentials` and `ext_uris`, which seems to make sense to me.

Reply:
Good catch, I was working against the tests so the ordering seemed logical. But you're right, those commits won't compile so I will reverse them.

Reply:
I agree with this. I removed login and logout to be managed by the caller. I also removed any reference to credentials and ext_uris since these do not need to be kept by the process. Since we no longer have a username stored I also removed generate_client_tr_id completely.

Reply:
Yeah, if decoupling the login call from `EppClient` then would be best to not have `login()` and `logout()` under `EppClient`. I don't have any problem with this and it makes more sense to me since the call to `logout()` had to be made explicit some time ago.

Initially I added them under `EppClient` because I wanted to support connection pooling, like in the SDK provided by Versign where this is done inside the lib, and where the connections in the pool stay active by making a `<poll>` request every few seconds to the EPP server, which can only be done when the `<login>` has already been performed.

Reply:
Yeah I agree that connection pooling would be nice. Or at least retry heuristics if the connection drops. 

I'm actually on the fence here. If we want to introduce more intelligent connection handling in the future, then keeping this login/logout abstraction could reduce work in the future since we would likely go through the client for that. Do you have any opinions here @djc?

Reply:
The pooling idea is still underdeveloped in how it should be implemented. For now I think it's best to go with this change and let users decide if they want to pool the `EppClient` instances. Maybe we can discuss any future plan for pooling as a separate feature.

Reply:
Until we actually implement pooling, it's probably better not to optimize for how it might be designed, so I think it makes sense to go through with these changes for now.

Reply:
Sounds good to me. I'll leave this as is.